src/main/java/duke/Main.java

```diff
@@ -0,0 +1,329 @@
+        String home = System.getProperty("user.dir");
+
+        Path folderPath = java.nio.file.Paths.get(home, "data");
+        Path filePath = java.nio.file.Paths.get(home, "data", "duke.txt");
+
+        if (!java.nio.file.Files.exists(folderPath)) {
+
+            try {
+                java.nio.file.Files.createDirectory(folderPath);
+            } catch (IOException e) {
+                System.out.println("The folder \"data\" is not created");
+            }
+        }
+        if (!java.nio.file.Files.exists(filePath)) {
+            try {
+                java.nio.file.Files.createFile(filePath);
+            } catch (IOException e) {
+                System.out.println("The file \"duke.txt\" is not created");
+            }
+        }
+        return filePath;
+    }
+
+    private static void printNumTask() {
+        System.out.println("Now you have " + tasks.size() + " tasks in the list.");
+    }
+
+    private static void deleteTask(int taskIndex) throws DeleteUndefinedTaskException {
+        if (taskIndex <= -1 || taskIndex >= tasks.size()) {
+            throw new DeleteUndefinedTaskException();
+        }
+
+        printLine();
+        System.out.println("Noted. I've removed this task: ");
+        System.out.println(tasks.get(taskIndex).toString());
+        tasks.remove(taskIndex);
+        printNumTask();
+        printLine();
+
+    }
+
+    private static void doneTask(int taskIndex) throws DoneUndefinedTaskException {
+        try {
+            tasks.get(taskIndex).markAsDone();
+        } catch (IndexOutOfBoundsException e) {
+            throw new DoneUndefinedTaskException();
+        }
+    }
+
+    private static void deleteOrDoneTask(String beginning, String command)
+            throws EmptyDoneException, DoneUndefinedTaskException, EmptyDeleteException, DeleteUndefinedTaskException {
+
+        if (beginning.equals(TASK_DONE)) {
+            if (command.substring(4).isBlank()) {
+                throw new EmptyDoneException();
+            }
+            try {
+                doneTask(Integer.parseInt(command.substring(5)) - 1);
+            } catch (NumberFormatException e) {
+                System.out.println("Please enter a number after done.");
+            }
+        } else {
+            if (command.substring(6).isBlank()) {
+                throw new EmptyDeleteException();
+            }
+            try {
+                deleteTask(Integer.parseInt(command.substring(7)) - 1);
+            } catch (NumberFormatException e) {
+                System.out.println("Please enter a number after delete.");
+            }
+        }
+    }
+
+    private static void respondMultiWordCommand(ArrayList<Task> tasks, String command) {
+        String[] words = command.split(" ");
+        if (words[0].equals(TASK_DONE) || words[0].equals(TASK_DELETE)) {
+            try {
+                deleteOrDoneTask(words[0], command);
+
+            } catch (DoneUndefinedTaskException | DeleteUndefinedTaskException e) {
```

Comment:
Perhaps you can always catch `DoneUndefinedTaskException` with an error message `e` instead?